### PR TITLE
BEAM-1228 Improve Regex

### PR DESCRIPTION
**Pull request by eljefe6a**
Improve Regex class.

### CODE REVIEW

**Comment by eljefe6a** [comment]
Ping @kennknowles

File: sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/Regex.java [hunk 1]
```diff
@@ -71,7 +138,46 @@ public static Matches matches(String regex, int group) {
...
+  /**
+   * Returns a {@link Regex.MatchesNameKV} {@link PTransform} that checks if the
+   * entire line matches the Regex. Returns the specified groups as the key and value as a
+   * {@link PCollection}.
+   *
+   * @param regex The regular expression to run
+   * @param keyGroupName The Regex group name to use as the key
+   * @param valueGroupName The Regex group name to use the value
+   */
+  public static MatchesNameKV matchesKV(String regex, String keyGroupName, String valueGroupName) {
```
Comment (kennknowles) [inline sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/Regex.java]: What about just saying the key group has to be named `key` and the value group has to be named `value`? Do we think that someone is going to have a regex lying around that they re-use for this transform? I bet they are always writing the regex just for this. So we don't really need to parameterize at all and don't need the `int` based version either.

Comment (eljefe6a) [inline sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/Regex.java]: Yes, they often have a regex that they've been using. Usually there's some kind of ETL regex that they're reusing. The `int` based version is used when the order for the KV is inverted in the regex. e.g. the value is group 1 and the key is group 2.

Comment (kennknowles) [inline sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/Regex.java]: OK if you say so then I'll buy it. LGTM

File: sdks/java/core/src/test/java/org/apache/beam/sdk/transforms/RegexTest.java [hunk 2]
```diff
@@ -94,7 +153,7 @@ public void testKVFindNone() {
   public void testMatches() {
 
     PCollection<String> output =
-        p.apply(Create.of("a", "x", "y", "z")).apply(Regex.matches("[xyz]"));
+            p.apply(Create.of("a", "x", "y", "z")).apply(Regex.matches("[xyz]"));
```
Comment (kennknowles) [inline sdks/java/core/src/test/java/org/apache/beam/sdk/transforms/RegexTest.java]: Formatting here actually. Use `google-java-format` if you can.

**Pull request closed without merging**